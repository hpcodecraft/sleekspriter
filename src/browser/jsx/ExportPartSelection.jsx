var ExportPartSelection = React.createClass({
  render: function() {
    var parts = [
      {name: 'spritesheet', el: 'Spritesheet as single image'},
      {name: 'oneframe', el:  <span>
                                Frame&nbsp;
                                <input type="number" value={this.props.frame} min={1} max={this.props.ui.frames.total} onChange={this.setFrame} />
                                &nbsp;/&nbsp;
                                {this.props.ui.frames.total} as image
                              </span>},
    ];

    return (
      <div>
        <h6>Export</h6>
        <ul>
          {parts.map(function(part) {
            var checked = part.name === this.props.part ? true : false;
            return (
              <li key={part.name}>
                <label>
                  <input type="radio" name="export-part" value={part.name} onChange={this.setPart}Â checked={checked} />
                  {part.el}
                </label>
              </li>
            );
          }, this)}
        </ul>
      </div>
    )
  },
  setPart: function(event) {
    channel.gui.publish('export.part.set', {part: event.target.value});
  },
  setFrame: function(event) {
    channel.gui.publish('export.frame.set', {frame: +event.target.value});
  },
  setAnimation: function(event) {
    channel.gui.publish('export.animation.set', {animation: event.target.value});
  },
});