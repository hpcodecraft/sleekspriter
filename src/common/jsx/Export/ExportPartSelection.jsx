var ExportPartSelection = React.createClass({
  getInitialState: function() {
    return {
      part: 'spritesheet',
      frame: 1,
      animation: null,
    }
  },
  render: function() {

    var parts = [
      {name: 'spritesheet', el: 'Spritesheet as single image'},
      {name: 'allframes', el: 'Every frame as single image'},
      {name: 'oneframe', el:  <span>
                                Frame&nbsp;
                                <input type="number" value={this.state.frame} min={1} max={this.props.editor.frames.total} onChange={this.setFrame} />
                                &nbsp;/&nbsp;
                                {this.props.editor.frames.total} as image
                              </span>},
      {name: 'animation', el: <span>
                                Animation&nbsp;
                                <select onChange={this.setAnimation}>
                                  {this.props.editor.animations.list.map(function(animation) {
                                    <option value={animation.name}>{animation.name}</option>
                                  }, this)}
                                </select>
                              </span>},
    ];

    return (
      <div>
        <h6>What to export</h6>
        <ul>
          {parts.map(function(part) {
            var checked = part.name === this.state.part ? true : false;
            return (
              <li key={part.name}>
                <label>
                  <input type="radio" name="export-part" value={part.name} onChange={this.setPart}Â checked={checked} />
                  {part.el}
                </label>
              </li>
            );
          }, this)}
        </ul>
      </div>
    )
  },
  setPart: function(event) {
    this.setState({part: event.target.value});
    channel.publish('export.part.set', {part: event.target.value});
  },
  setAnimation: function(event) {
    this.setState({animation: event.target.value});
    channel.publish('export.animation.set', {animation: event.target.value});
  },
  setFrame: function(event) {
    this.setState({frame: +event.target.value});
    channel.publish('export.frame.set', {frame: +event.target.value});
  },
});