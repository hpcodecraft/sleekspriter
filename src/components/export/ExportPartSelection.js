import React from "react";
import { clamp } from "lodash";
import { t } from "../../utils";

const ExportPartSelection = ({ frame, part, totalFrames, setFrame, setPart }) => {
  const parts = [
    { name: "spritesheet", el: t("Spritesheet as single image") },
    { name: "oneframe", el:  <span>
                              Frame&nbsp;
                              <input type="number" value={frame} min={1} max={totalFrames} onChange={(e) => setFrame(clamp(+e.target.value, 1, totalFrames))} />
                              &nbsp;/&nbsp;
                              {totalFrames} as image
                            </span> }
  ];

  return (
    <div>
      <h6>{t("Export")}</h6>
      <ul>
        {parts.map(function(p) {
          var checked = p.name === part ? true : false;
          return (
            <li key={p.name}>
              <label>
                <input type="radio" name="export-part" value={p.name} checked={checked} onChange={(e) => setPart(e.target.value)} />
                {p.el}
              </label>
            </li>
          );
        }, this)}
      </ul>
    </div>
  );
};

export default ExportPartSelection;
