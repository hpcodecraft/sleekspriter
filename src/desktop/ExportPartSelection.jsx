var ExportPartSelection = React.createClass({
  render: function() {
    var parts = [
      {name: 'spritesheet', el: 'Spritesheet as single image'},
      {name: 'oneframe', el:  <span>
                                Frame&nbsp;
                                <input type="number" value={this.props.frame} min={1} max={this.props.editor.frames.total} onChange={this.setFrame} />
                                &nbsp;/&nbsp;
                                {this.props.editor.frames.total} as image
                              </span>},
      {name: 'allframes', el: 'Every frame as single image'},
    ];

    if(this.props.editor.animations.list.length > 0) {
      parts.push({
        name: 'animation',
        el: <span>
              Animation&nbsp;
              <select onChange={this.setAnimation}>
                {this.props.editor.animations.list.map(function(animation) {
                  return( <option key={animation.name} value={animation.name}>{animation.name}</option> )
                }, this)}
              </select>
            </span>
      });
    }

    return (
      <div>
        <h6>Export</h6>
        <ul>
          {parts.map(function(part) {
            var checked = part.name === this.props.part ? true : false;
            return (
              <li key={part.name}>
                <label>
                  <input type="radio" name="export-part" value={part.name} onChange={this.setPart}Â checked={checked} />
                  {part.el}
                </label>
              </li>
            );
          }, this)}
        </ul>
      </div>
    )
  },
  setPart: function(event) {
    channel.publish('export.part.set', {part: event.target.value});
  },
  setFrame: function(event) {
    channel.publish('export.frame.set', {frame: +event.target.value});
  },
  setAnimation: function(event) {
    channel.publish('export.animation.set', {animation: event.target.value});
  },
});