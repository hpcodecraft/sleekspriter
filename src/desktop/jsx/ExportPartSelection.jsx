// Flux: done, editor: done
var ExportPartSelection = React.createClass({
  mixins: [FluxMixin],
  render: function() {
    var parts = [
      {name: 'spritesheet', el: 'Spritesheet as single image'},
      {name: 'oneframe', el:  <span>
                                Frame&nbsp;
                                <input type="number" value={this.props.ui.export.frame} min={1} max={this.props.ui.frames.total} onChange={this.setFrame} />
                                &nbsp;/&nbsp;
                                {this.props.ui.frames.total} as image
                              </span>},
      {name: 'allframes', el: 'Every frame as single image'},
    ];

    if(this.props.file.animations.length > 0) {

      var id = this.props.ui.export.animation === null ? this.props.file.animations[0].id : this.props.ui.export.animation,
          animation = storeUtils.animations.getById(id),
          runtime = Math.round(1000/animation.fps * animation.frames.length);

      parts.push({
        name: 'animation',
        el: <span>
              Animation&nbsp;
              <select onChange={this.setAnimation} defaultValue={id}>
                {this.props.file.animations.map(function(animation) {
                  return( <option key={animation.id} value={animation.id}>{animation.name}</option> )
                }, this)}
              </select>
              <i className="animation-info">
                {animation.frames.length} frames, {animation.fps} fps<br />
                Runtime: {runtime} ms
              </i>
            </span>
      });
    }

    return (
      <div>
        <h6>Export</h6>
        <ul>
          {parts.map(function(part) {
            var checked = part.name === this.props.ui.export.part ? true : false;
            return (
              <li key={part.name}>
                <label>
                  <input type="radio" name="export-part" value={part.name} onChange={this.setPart}Â checked={checked} />
                  {part.el}
                </label>
              </li>
            );
          }, this)}
        </ul>
      </div>
    )
  },
  setPart: function(event) {
    this.getFlux().actions.exportPart(event.target.value);
  },
  setFrame: function(event) {
    this.getFlux().actions.exportFrame(event.target.value);
  },
  setAnimation: function(event) {
    this.getFlux().actions.exportAnimation(event.target.value);
  },
});