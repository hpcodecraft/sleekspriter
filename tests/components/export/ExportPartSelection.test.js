import ExportPartSelection from "components/export/ExportPartSelection";

describe("ExportPartSelection", () => {
  let props, jsx, wrapper;
  beforeEach(() => {
    props = {
      frame: 1,
      part: "spritesheet",
      totalFrames: 4,
      setPart: jest.fn(),
      setFrame: jest.fn(),
    };
    jsx = <ExportPartSelection {...props} />;
    wrapper = shallow(jsx);
  });

  it("should render correctly", () => {
    expect(render(jsx)).toMatchSnapshot();
  });

  describe("radio buttons", () => {
    it("fire setPart handler when clicked", () => {
      wrapper
        .find('input[value="frame"]')
        .simulate("change", { target: { value: "frame" } });
      expect(props.setPart).toHaveBeenCalledWith("frame");
    });
  });

  describe("frame number input", () => {
    it("fires setFrame handler on change", () => {
      wrapper
        .find('input[type="number"]')
        .simulate("change", { target: { value: 3 } });
      expect(props.setFrame).toHaveBeenCalledWith(3);
    });

    it("clamps the setFrame handler to a minimum of 1", () => {
      wrapper
        .find('input[type="number"]')
        .simulate("change", { target: { value: -1 } });
      expect(props.setFrame).toHaveBeenCalledWith(1);
    });

    it("clamps the setFrame handler to a maximum of the total frames", () => {
      wrapper
        .find('input[type="number"]')
        .simulate("change", { target: { value: 9001 } });
      expect(props.setFrame).toHaveBeenCalledWith(props.totalFrames);
    });
  });
});
