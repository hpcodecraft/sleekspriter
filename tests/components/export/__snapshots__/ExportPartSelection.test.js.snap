// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportPartSelection should render correctly 1`] = `"<div><h6>Export</h6><ul><li><label><input type=\\"radio\\" name=\\"export-part\\" value=\\"spritesheet\\" checked=\\"\\"/>Spritesheet as single image</label></li><li><label><input type=\\"radio\\" name=\\"export-part\\" value=\\"frame\\"/><span>Frame <input type=\\"number\\" value=\\"1\\" min=\\"1\\" max=\\"4\\"/> / 4 as image</span></label></li></ul></div>"`;
